### PR TITLE
Remove Merge From Scene

### LINKED ISSUE #32039: Merge from scene confusion (maybe bug)
maybe this same like this, my less complicated https://github.com/godotengine/godot/issues/31744

**Godot version:**
3.1.1 stable

**OS/device including version:**
Editor Linux 

**Issue description:**
On Merge from scene, created in editor resources shared only for current session, reopening editor unlink them all from each other.

**Steps to reproduce:**
*Create Root Node*: 2D scene (two scene, two editor tabs), save them.

*first tab*(scene1): create Sprite, on first scene, create new Gradient texture->new Gradient, save

*second Tab*(scene2): press Merge from scene, select scene1, Sprite

now editing(change any color) Gradient on Sprite in any of tabs(Scenes) make it same change to second Sprite Gradient.

Save everything, close Editor.

Reopen Editor: now changing Gradient do not set same value to other Sprite in other scene.

**The problem is** - it exist in all materials(particle material/spatial material/other), Gradient and Curve

Copying objects form scene to scene with material that has saved *local* created elements, global editing does not work after Editor restart.(I did not knew that and now need to remake 6 scenes with 6 copyes on each 2 material per object to make it work again in my project)

**Minimal reproduction project:**
not needed

### PR DESCRIPTION
**Pull request by KoBeWi**
This PR removes Merge From Scene, which is useless after #34892 was merged.
Hopefully I didn't miss anything.

Resolves #32039 (copy-paste doesn't have this problem)

### CODE REVIEW

**Comment by RandomShaper** [comment]
Makes sense. However, it's a bit frightening, isn't it? 😝

**Review by RandomShaper — APPROVED** [review]

**Review by akien-mga — APPROVED** [review]
I'm a bit concerned that it can appear as a workflow breakage for some users, at least until used to being able to copy paste... but there's still time to try it out until 4.0 is released, so let's remove it and see how the feedback is.

**Comment by akien-mga** [comment]
Thanks!

**Comment by KoxaKoxama** [comment]
@akien-mga @KoBeWi @RandomShaper it was a workflow breakage for me! As I understand, I can now just select all the files in the scene and copy them to another one? Is it a safe workflow?

**Comment by KoBeWi** [comment]
Yes, it's even safer than the merge from scene, because it automatically makes resources unique when necessary.

**Comment by KoxaKoxama** [comment]
@KoBeWi @RandomShaper @reduz The only problem I can't figure out is why the size of my scene is getting so big. Every time I copy a node from another scene containing a mesh, it is completely copied to the new scene, literally all mesh coordinates go directly to the tscn file in PoolByteArray, is this a normal behavior? Why not instantiate the original mesh file? 
If I have a large enough number of detailed mesh objects that I want to copy into the main scene to build a level (but I don't want to copy them as separate scenes, I need the mesh directly) why is the mesh literally copied into the scene file? My tscn file is under 400mb, it's very strange...

**Comment by KoBeWi** [comment]
What version are you using? It sounds like an old bug that should be fixed

**Comment by KoxaKoxama** [comment]
This is in 3.5.1 and 4.0b2
Here is my issue #66991

**Comment by PetBom** [comment]
@KoBeWi 

I just noticed that "Merge From Scene" is missing from Godot 4 (Beta 8), and this PR confirms that it indeed is so. 

I would challenge the initial assumption of this PR that Copy/Paste will replace the "Merge from scene" use cases. 

"Merge from scene" is able to pick resources from a scene file (.tscn, .glb...) without actually importing the scene. This is a really useful feature that allow you to pick say just a mesh from a file for replacement without having to import/re-import a whole scene. I would go as far as suggesting that this is the primary use case for "Merge from scene"

This is a quite widely used use case in my workflow. I see no way that could be done with Copy/Paste functionality, but maybe I'm missing something.

**Comment by KoBeWi** [comment]
> "Merge from scene" is able to pick resources from a scene file (.tscn, .glb...) without actually importing the scene. This is a really useful feature that allow you to pick say just a mesh from a file for replacement without having to import/re-import a whole scene. I would go as far as suggesting that this is the primary use case for "Merge from scene"

I think I don't understand. When you copy a scene instance with any external references, the references are left unchanged. Only built-in references are copied (but save branch as scene does the same, except it's bugged).

Did you try if copy-paste works with your use-case or are just asking theoretically?

**Comment by Zireael07** [comment]
> "Merge from scene" is able to pick resources from a scene file (.tscn, .glb...) without actually importing the scene. This is a really useful feature that allow you to pick say just a mesh from a file for replacement without having to import/re-import a whole scene. I would go as far as suggesting that this is the primary use case for "Merge from scene"

I must say I don't understand either. I've used copy and paste to do exactly this (extract a single mesh instance from a glb file)

**Comment by PetBom** [comment]
Ok. Maybe I'm explaining badly. Let me try again. Here's an example 3.5 scene to show my workflow:

image

Notice that I do not have an instance of the tree.glb in the project node tree. It is imported but not instantiated anywhere in the  project. If i now right click the project node tree and use "Merge from scene":

image

Select file to import from...
image

Select node(s) to import...
image

And the selected node gets merged into the project node tree.
image

So this is the use case I'm trying to describe, and that I believe is missing from the Copy/Paste solution. I see no way to do the same maneuver without actually creating an instance of the tree.glb in the project node tree. 

I could see ways that you could implement something similar using Copy/Paste instead. If you implemented Copy from the Advanced Import view Node tree.

As I've manly used the 3.5 "Merge from scene" to actually pick up different meshes for bone deformation using the same skeleton (Multiple versions o character created in blender as separate .glbs let me pick up the desired mesh and merge it in to the instantiated version of the "master" scene for that character) i could see how this specific workflow could be realized in 4.0 using something like Animation Retargeting. But i still think there are many other usecases where "picking up nodes" from file is really useful.

**Comment by Zireael07** [comment]
That's exactly what I did with copy and paste. Open the glb, copy and paste, voila! No need to "add to project node tree"

**Comment by PetBom** [comment]
Alright. I think I get it now! :) So to adapt the workflow to 4.0 you would go:

Right click tree.glb in File System and select "New Inherited Scene"... 
image

Right click "Copy" the selected node in the Inherited scene node tree...
image

Then "Paste" into the project node tree...
image

And, voila! There we are!
image

And then just discard the unsaved scene tab.

Yeah that works. For users accustomed to "Merge from scene" (like me) it's not super obvious but once explained it makes sense. Thanks for clarifying. I'll just leave this here if someone else stumbles along with same question ;)
